service/resource/configmap/resource.go

```diff
@@ -206,7 +206,7 @@ func (r *Resource) ProcessCreateState(ctx context.Context, obj, createState inte
 	}
 
 	// Create the config maps in the Kubernetes API.
-	if configMapsToCreate != nil {
+	if len(configMapsToCreate) != 0 {
```

Comment:
This is a bit more safe for the resources that deal with multiple entities if themselves.